authentication/middleware.py

```diff
@@ -30,7 +30,7 @@ def process_exception(self, request, exception):
             url = self.get_redirect_uri(request, exception)
 
             if url:  # noqa: RET503
-                url += ("?" in url and "&" or "?") + "message={0}&backend={1}".format(  # noqa: UP030, UP032
+                url += ("?" in url and "&" or "?") + "message={}&backend={}".format(  # noqa: UP032
```

Comment:
The reason `pyupgrade` did not convert this into an f-string is likely due to its design choice to prioritize readability. According to the pyupgrade [documentation](https://github.com/asottile/pyupgrade?tab=readme-ov-file#f-strings), it intentionally avoids converting expressions into `f-strings` if doing so would make the expression longer or if the substitution parameters are sufficiently complicated, as this can decrease readability.

In this case, the expression might have been considered sufficiently complex, or the transformation into an f-string might have made it less readable, which is why pyupgrade retained the .format() syntax.

However, if you believe an f-string would improve clarity in this case, I'm open to making the change manually. What do you think?

This holds true for similar comments you mentioned regarding `f-string` where `pyupgrade` intentionally avoids converting expressions into `f-strings`.